File: components/Button/atlas_button.dart
import 'package:flutter/material.dart';
import 'package:atlas_ui/atlas_tokens.dart';

enum AtlasButtonVariant { primary, secondary, danger }

class AtlasButton extends StatelessWidget {
  const AtlasButton({
    super.key,
    required this.label,
    required this.onPressed,
    this.variant = AtlasButtonVariant.primary,
  });

  final String label;
  final VoidCallback onPressed;
  final AtlasButtonVariant variant;

  @override
  Widget build(BuildContext context) {
    return FilledButton(
      style: FilledButton.styleFrom(
        minimumSize: const Size.fromHeight(44),
        shape: RoundedRectangleBorder(borderRadius: AtlasTokens.controlRadius),
        backgroundColor: AtlasTokens.buttonColor(variant),
      ),
      onPressed: onPressed,
      child: Text(label),
    );
  }
}

File: components/Button/AtlasButton.kt
package com.atlas.ui.button

import androidx.compose.material3.Button
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable

enum class AtlasButtonVariant {
  Primary,
  Secondary,
  Danger
}

@Composable
fun AtlasButton(
  label: String,
  variant: AtlasButtonVariant = AtlasButtonVariant.Primary,
  onClick: () -> Unit,
) {
  Button(
    onClick = onClick,
    shape = AtlasButtonTokens.shape,
    colors = AtlasButtonTokens.colors(variant),
  ) {
    Text(label)
  }
}
